flo-runner/src/main/java/com/spotify/flo/context/ForkingEvalContext.java

```diff
@@ -0,0 +1,344 @@
+        final Path errorFile = tempdir.resolve("error");
+
+        final String home = System.getProperty("java.home");
+        final String classPath = System.getProperty("java.class.path");
+        final Path java = Paths.get(home, "bin", "java").toAbsolutePath().normalize();
+
+        try {
+          PersistingContext.serialize(value, closureFile);
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to serialize closure", e);
+        }
+
+        final ProcessBuilder processBuilder = new ProcessBuilder()
+            .command(
+                java.toString(),
+                "-cp", classPath,
+                Trampoline.class.getName(),
+                closureFile.toString(),
+                resultFile.toString(),
+                errorFile.toString())
+            .directory(workdir.toFile());
+
+        final Process process;
+        try {
+          process = processBuilder.start();
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+
+        // TODO: Log output
+
+        // Plan A
+        // ======
+        // 1. Set up logging in the child process to emit structured json (with severity and timestamps etc) to a file
+        // 2. Make the parent tail the log file, parse the json messages, decorate with task metadata and re-emit the
+        //    log messages. Note that re-emitting the log messages through the parent process slf4j logger would involve
+        //    figuring out how to keep the original log message metadata (e.g. timestamp, logger, thread, etc). Maybe
+        //    using the LocationAwareLogger interface. Printing the log messages directly to stderr might be more
+        //    straightforward. flo does not pull in a logging implementation, we'd have to pull one in (e.g. logback)
+        //    and set it up to do structured logging in the child process. Alternatively roll our own.
+        // 3. Also tail child process std{out,err} for unstructured output and log each line using the parent process
+        //    logger, taking care to not choke on huge lines and non-text output.
+
+        // Plan B
+        // ======
+        // Outsource the problem completely to the logger implementation provided by the user:
+        // 1. No special setup for the child process, it just logs to std{out,err} as configured by user.
+        // 2. Parent process tails process std{out,err} and re-logs each line using the slf4j logger. Any structured
+        //    log messages now end up wrapped in the message field of the log message.
+        // 3. The parent process logger inspects all log messages and bypasses normal processing for any (wrapped)
+        //    structured log messages, just decorating it with task metadata and emitting it as is.
+
+        // Note:
+        // * Child process does not need grpc context
+        // * Tempted at this point to run the child task in a container and let GKE deal with the logs
```

Comment:
A scheduler inside a scheduler. :D
We can as well abandon orchestration and let Styx handle it.

Reply:
\o/

Reply:
Make flo just emit the dag and let styx run the tasks?